subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/CompileWithAnnotationProcessingBuildOperationType.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.compile;
+
+import org.gradle.internal.operations.BuildOperationType;
+
+import java.util.Map;
+
+public class CompileWithAnnotationProcessingBuildOperationType implements BuildOperationType<CompileWithAnnotationProcessingBuildOperationType.Details, CompileWithAnnotationProcessingBuildOperationType.Result> {
+
+    public interface Details {
+        String getTaskPath();
```

Comment:
I don't think we need this, since the operation will be a child of the compile task operation, so the relationship is clear.

@ldaley Would you agree or do you include paths in other places to simplify the plugin logic?

Reply:
If it’s _always_ a child of the task, can be omitted. General strategy is less is more.

Reply:
Removed in f1ecfc146c4.